# SYCL Academy

## Exercise 3: Handling Errors

---

In this exercise, you will learn how to handle synchronous and asynchronous
exceptions.  Specifically, you will introduce a few errors and catch them
using standard C++ try catch.  Of course, asynchornous errors are caught
by the SYCL runtime and made available to us to handle with standard C++
mechanisms which only support a synchronous model directly.

We assume you have your environment set up to compile, since we did that in Exercise 01.
Ask for help if you have not completed Exercise 01 - we are here to help!

This exercise uses the solution from Exercise 02 as it's basis, which was based on Exercise 01.
It should look familiar.
If you have extra time, we hope you'll explore making more additions or changes.

---

### 1.) JAMES WORKING ON THIS...



TBD
TBD
TBD
TBD
TBD
TBD
TBD
TBD
TBD
TBD






#### Build And Execution Hints

For DPC++ (using the Intel DevCloud):
```sh
cd ~/syclacademy/Code_Exercises/Exercise_03_Error_Handling
icpx -fsycl -o sycl-ex-3 source.cpp
cp ../Images/goldfish.png  .
./sycl-ex-3 goldfish.png
```
